### PR TITLE
add support for rootdisk as data_disk_type and change default drbd fencing to sbd

### PR DESCRIPTION
**Pull request by melzer-b1**
Deployment failed on using rootdisk as data source.

### CODE REVIEW

**Review by yeoldegrove — CHANGES_REQUESTED** [review]
@melzer-b1 Thanks for adding this!
Could you please add an example line for `rootdisk` in `terraform.tfvars.example`?

**Comment by melzer-b1** [comment]
added rootdisk example in terraform.tfvars.example

**Review by arbulu89 — COMMENTED** [review]
Hi guys,

I found out now that we create a NFS service in the bastion host for openstack. I don't like much this idea, but well, I guess it is fine.

I would suggest to put some comments in the salt code saying that this only applies to openstack

File: salt/bastion/init.sls [hunk 1]
```diff
@@ -1,9 +1,10 @@
-{% if grains.get('monitoring_enabled', False) or grains.get('bastion_data_disk_type') in ['ephemeral', 'volume'] %}
+{# 'rootdisk' is used for nfs server on openstack #}
```
Comment (yeoldegrove) [inline salt/bastion/init.sls:1]: `rootdisk` AND `ephemeral` AND `volume` are only used on openstack.
Maybe rephrase it like "NFS server for `/sapinst` on bastion is only available on openstack." ?

**Review by yeoldegrove — APPROVED** [review]

**Comment by yeoldegrove** [comment]
@arbulu89 good to merge now from your side?

File: salt/bastion/sapinst.sls [hunk 2]
```diff
@@ -10,6 +10,9 @@ nfs-kernel-server:
         attempts: 3
         interval: 15
 
+{# NFS server used for '/sapinst' on bastion is only available on openstack. #}
```
Comment (arbulu89) [inline salt/bastion/sapinst.sls:13]: Do we really need this `jinja` kind of comments?
Using simple `# my comment` should work too

**Comment by arbulu89** [comment]
> @arbulu89 good to merge now from your side?

Yes. I don't personally like this `jinja` kind of comments (mostly, because the rest of the code uses normal comments).
But it is acceptable of course.
Up to you if you want to change them or not

**Comment by melzer-b1** [comment]
My thoughts by using jinja2 comments are: First we can distinguish comments for jinja2 blocks from ordinary salt comments.
Secondary there is a jinja2 feature where you can write a expresion "{% if … }" as "# if …" so i wanted to avoid collisions.
We found some other cases of jinja2 comments in the code base so we decided to use them.

If you think this conflicts to your coding style I can change it to the other type of comment.
